### ISSUE TITLE
Not Working Anymore

### ISSUE THREAD

**Issue by volomike** [bogkonstantin/android_income_sms_gateway_webhook#46]
I used to have this working on my old Samsung Galaxy phone. I tried it on a new Walmart Tracfone Samsung A13 (which, instead of SMS and MMS, it has now this new RCS technology), and it's not forwarding. I tried the webhook.site like you recommended for troubleshooting, and doublechecked the sending URL, and that didn't help. I also rebooted my phone and did more tests and it's not sending anything.

**Comment by volomike** [comment]
Okay, I can confirm. When I took another phone and set it to not do "Advanced Messaging" or "Chat Mode" and simply let it send ordinary "text messages" and "multimedia messages", that translates it to sending a true SMS message when I went to send some text. It doesn't matter what the gateway phone is set on -- it receives this as an SMS message and then this Android tool will automatically forward. So, evidently, it does not yet support the new RCS system, only the SMS system. Since most services that send SMS alerts or do 2FA alerts only use SMS, this won't be a problem for now and this tool does forward properly.

Any possibility of a new feature to support RCS mode? Not a huge issue for now, in my opinion.

**Comment by prabhuon** [comment]
yes, this also not work if the mobile connected with voice over WiFi, any update when this trigger can be implemented

**Comment by LukasLewandowski** [comment]
Hi, any plans to implement this?

**Comment by bogkonstantin** [comment]
Hi, thanks for the report.

Unfortunately, RCS support isn't possible in an app like this. RCS is a completely different transport from SMS — it doesn't trigger the `SMS_RECEIVED` broadcast this app relies on, and **Android has no public API for third-party apps to receive RCS messages.** The only RCS reception API is hidden (`@hide`) and locked to a Samsung-only allowlist, and the public `ImsRcsManager` only exposes capability/presence info, not message content. RCS messages also aren't stored in the public Telephony content provider.

**Workaround:** disable RCS in Google Messages (Settings → RCS chats → Turn off RCS chats). Your messages will then arrive as normal SMS, which this app forwards correctly.

Documented in 0e0aafa. Closing as not feasible.
